content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,149 @@
+profiles. Those seccomp profiles can be enabled on individual workloads,
+[a security tutorial is available](/docs/tutorials/security/seccomp/). In
+addition, the [Kubernetes Security Profiles Operator](https://github.com/kubernetes-sigs/security-profiles-operator)
+is a project to facilitate the management and use of seccomp in clusters.
+
+Please note that Docker has been using
+[a default seccomp profile](https://docs.docker.com/engine/security/seccomp/)
+to only allow a restricted set of syscalls but Kubernetes is still not
+confining workloads by default. Fortunately,
+[Seccomp Default](/blog/2021/08/25/seccomp-default/), a new alpha feature to
+use a default seccomp profile for all workloads can now be enabled and tested.
+
+## Pod placement
+
+Pods that could be considered different tiers of sensitivity, for example an
+application pod and the Kubernetes API server, should be deployed onto separate
+nodes. This separation should be enforced to prevent pods accidentally being
+deployed onto the same node. This could be enforced with the following features:
+
+- [Node Selectors](/docs/concepts/scheduling-eviction/assign-pod-node/) are
+  key-value pairs as part of the pod specification that specifies which nodes
+  to deploy onto. These can be enforced at the namespace and cluster level with
+  the
+  [PodNodeSelector](/docs/reference/access-authn-authz/admission-controllers/#podnodeselector)
+  admission controller.
+- [PodTolerationRestriction](/docs/reference/access-authn-authz/admission-controllers/#podtolerationrestriction)
+  is an admission controller that allows administrators to add permitted
+  [tolerations](/docs/concepts/scheduling-eviction/taint-and-toleration/)
+  within a namespace. Pods within that namespace may only utilise those
+  specified in the namespace.
+
+## Secrets
+
+- Secrets required for pods should be stored within Kubernetes secrets as
+  opposed to alternatives such as ConfigMap.
+- Pods with secrets should have these automatically mounted through volumes.
+  This would allow adding secrets into the pod as environment variables or
+  files on the filesystem. This should be done preferentially as compared to
+  providing the pods service account RBAC access to the secret.
+- Secret resources stored within etcd should be
+  [encrypted at rest](/docs/tasks/administer-cluster/encrypt-data/).
+- Service accounts should not be mounted into pods that do not require them.
+  This can be configured by setting
+  [`automountServiceAccountToken`](/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server)
+  to `false` either within the service account to apply throughout the
+  namespace or specifically for a pod.
+- Kubernetes v1.22 and above - Utilise
+  [Bound Service Accounts](/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
+  for time-bound service account credentials.
+
```

Comment:
I think this section is very specific, and if you use a cloud provider for your Kubernetes cluster, as far as I know you cannot change the settings of the Kubelet at all. So personally I would not add this section to the checklist. - But this is just my opinion.

Reply:
If I remember correct, some cloud vendors have this admission controller configured, but its not often used.
I also agree with your opinion, its specific and depends a bit on the Kubernetes 'vendor'.

Reply:
I personally think we should still add it, not everyone deploys Kubernetes with a cloud provider. Maybe we add a sub section somewhere specifically for what security considerations should be different when using a cloud provider based cluster.

Reply:
If it's generic and general enough I think it might be adapted for a section on this checklist! Being limited by what cannot be modified because of X or Y cloud provider offer is not a good idea because it can greatly vary between different distributions. It might be a better idea to suppose a "vanilla" distribution and add information that might be relevant to *most* of the managed solutions.

Reply:
Hey @p4ck3t0, I added a section about admission controllers thanks to @Skybound1 here https://github.com/kubernetes/website/pull/33992/commits/48c960490bf8ec7268cd193da7910d86c1e98e4a, if you want, you can add precisions about `NodeRestriction` here.

Reply:
Resolved by https://github.com/kubernetes/website/pull/33992/commits/2672d212fa9585eea7acda2028c2757461ecd2f7.